src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -740,6 +752,9 @@ static Acl buildAcl(ResultSet rs, String delim, String namespace)
         denyGroups.addAll(getGroupPrincipalsFromResultSet(rs,
             GsaSpecialColumns.GSA_DENY_GROUPS, delim, namespace));
       }
+      if (rsNext == ResultSetNext.STOP_NEXT) {
```

Comment:
Please help me understand the constraint that ACL coming from singleDocContentSql imposes that requires this statement.

Reply:
We shouldn't be calling rs.next() when processing ACLs from singleDocContentSql, or else we would be moving to next or end of result set and wouldn't be able to get content from result set.